### ISSUE TITLE
Error in seastar.pc written by build.ninja (hence configure.py)

### ISSUE THREAD

**Issue by brianthelion** [scylladb/seastar#40]
Currently, `seastar.pc` gets written as follows:

```
-e Name: Seastar
URL: http://seastar-project.org/
Description: Advanced C++ framework for high-performance server applications on modern hardware.
Version: 1.0
Libs: -L/home/br/Dev/foo/seastar/build/release -Wl,--whole-archive -lseastar -Wl,--no-whole-archive -g -Wl,--no-as-needed   -fvisibility=hidden -pthread  -laio -lboost_program_options -lboost_system -lstdc++ -lm -lboost_unit_test_framework -lboost_thread -lcryptopp -lrt -lxenstore -lhwloc -lnuma -lpciaccess -lxml2 -lz 
Cflags: -std=gnu++1y -g  -Wall -Werror -fvisibility=hidden -pthread -I/home/br/Dev/foo/seastar -I/home/br/Dev/foo/seastar/build/release/gen   -DHAVE_XEN -DHAVE_HWLOC -DHAVE_NUMA  -O2
```

Note the "-e" in the first line. The result when running `pkg-config` is:

``` bash
$ pkg-config --cflags --libs ../../seastar/build/release/seastar.pc
Package '/seastar' has no Name: field
```

The "-e" is due to [line 382 in configure.py](https://github.com/cloudius-systems/seastar/blob/2520f81368aebcf63e777ec515c1ac0f872b5283/configure.py#L382).

**Comment by nyh** [comment]
In Ubuntu, /bin/sh is not bash, it is dash. Because of decades-old historic
argument, dash and bash differ in their implementation of the "echo"
builtin - bash supports "-e" while "dash" does not.

The easiest solution is to replace the call to "echo" by a call to
"/bin/echo". On all modern systems, "/bin/echo" is GNU echo, which supports
"-e".


On Thu, Apr 16, 2015 at 6:34 AM, brianthelion <notifications@github.com>
wrote:

> Currently, seastar.pc gets written as follows:
>
> -e Name: Seastar
> URL: http://seastar-project.org/
> Description: Advanced C++ framework for high-performance server applications on modern hardware.
> Version: 1.0
> Libs: -L/home/br/Dev/foo/seastar/build/release -Wl,--whole-archive -lseastar -Wl,--no-whole-archive -g -Wl,--no-as-needed   -fvisibility=hidden -pthread  -laio -lboost_program_options -lboost_system -lstdc++ -lm -lboost_unit_test_framework -lboost_thread -lcryptopp -lrt -lxenstore -lhwloc -lnuma -lpciaccess -lxml2 -lz
> Cflags: -std=gnu++1y -g  -Wall -Werror -fvisibility=hidden -pthread -I/home/br/Dev/foo/seastar -I/home/br/Dev/foo/seastar/build/release/gen   -DHAVE_XEN -DHAVE_HWLOC -DHAVE_NUMA  -O2
>
> Note the "-e" in the first line. The result when running pkg-config is:
>
> $ pkg-config --cflags --libs ../../seastar/build/release/seastar.pc
> Package '/seastar' has no Name: field
>
> The "-e" is due to line 382 in configure.py
> <https://github.com/cloudius-systems/seastar/blob/2520f81368aebcf63e777ec515c1ac0f872b5283/configure.py#L382>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cloudius-systems/seastar/issues/40>.
>



-- 
Nadav Har'El
nyh@cloudius-systems.com

**Comment by nyh** [comment]
This was fixed by commit 3c8a5a89
So this issue should be closed.

**Comment by jvanz** [comment]
@nyh this issue seems to be fixed. But it's still open.
